thread/thread.cpp

```diff
@@ -534,14 +540,7 @@ namespace photon
             vcpu = current->get_vcpu();
             (plock = &vcpu->runq_lock) -> foreground_lock();
         }
-        mutable bool update_current = false;
-        void set_current(thread* th) const {
```

Comment:
There's no longer the need to set/switch the thread-local variable CURRENT as part of context switching, given that it has become a fiber-local varible.